torch/testing/_creation.py

```diff
@@ -81,7 +82,7 @@ def make_tensor(
 
     Raises:
         ValueError: If ``requires_grad=True`` is passed for integral `dtype`
-        ValueError: If ``low > high``.
+        ValueError: If ``low >= high``.
```

Comment:
With this change I'm off two minds:

- Is the current behavior a bug? What does it mean to request values drawn from the interval `[x, x)`? That seems like nonsense. So maybe we're just detecting that case properly now, and fixing the bug that we ever accepted it.
- Is the current behavior a deprecation? If so, we should update the documentation following [the BC policy](https://github.com/pytorch/pytorch/wiki/PyTorch's-Python-Frontend-Backward-and-Forward-Compatibility-Policy), and add a `.. deprecated::` directive. 

Either approach seems OK to me

Reply:
Given that

1. we documented that we only error for `low > high` and
2. `low == high` even works for floating and complex dtypes,

I think we should go through a proper deprecation cycle. Thanks for the heads-up for the docs. Will add it.

Reply:
![Screenshot from 2023-03-21 17-13-22](https://user-images.githubusercontent.com/6849766/226672482-2007c0c7-17a5-4502-95d4-4166646930d5.png)